specifyweb/workbench/upload/upload_plan_schema.py

```diff
@@ -217,14 +232,17 @@
 
 def parse_plan_with_basetable(collection, to_parse: Dict) -> Tuple[Table, Uploadable]:
     base_table = datamodel.get_table_strict(to_parse['baseTableName'])
-    return base_table, parse_uploadable(collection, base_table, to_parse['uploadable'])
+    extra = {}
+    if 'treeId' in to_parse.keys():
+        extra['treedefid'] = to_parse['treeId']
```

Comment:
> Here is an example of the upload plan I designed the back-end to handle:
> 
> ```json
> {
>   "baseTableName": "taxon",
>   "treeId": 1, # base tree table, ranks default to this tree when no treeId is given with the rank
>   "uploadable": {
>     "treeRecord": {
>       "ranks": {
>         "Kingdom": {
>           "treeNodeCols": {
>             "name": "Kingdom",
>             "author": "Kingdom - Author"
>           },
>           "treeId": 1 # the treedefid of the rank
>         },
>         "Phylum": {
>           "treeNodeCols": {
>             "name": "Phylum",
>             "author": "Phylum - Author"
>           },
>           "treeId": 1 # the treedefid of the rank
>         },
>         "Kingdom Test": {
>           "treeNodeCols": {
>             "name": "Kingdom Test",
>             "author": "Kingdom Test - Author"
>           },
>           "treeId": 2 # the treedefid of the rank
>         }
>       }
>     }
>   }
> }
> ```
> 
> The treeId it the id of the base table taxon id, and within the ranks, the treedefid is the treedefid of the rank. If the treedefid is not given for the rank, the taxontreedefid is used, or we just search for the rank by name and throw an error if there are duplicates. On the backend, I'm allowing for taxonTreeId and treedefid to be optional, so the backend will try and infer what they should be, and then will throw an error if it cannot determine the treeid. If inferences are made, the upload plan is edited on the backend to add those values.
> 
> I experimented with adding definition in the treeNodeCols, but the front-end threw errors about putting a relationship in the treeNodeCols. Let me know if we should try and get that to work, rather than using treedefid outside of treeNodeCols.
> 
> Let's discuss further in dev meeting.